### ISSUE TITLE
Hierarchy types

### ISSUE THREAD

**Issue by ocervell** [terraform-google-modules/cloud-foundation-fabric#12]
Would be nice to have some other commonly used hierarchies for GCP folders. 

We could have the following hierarchy types:

- **[exists]** Environment folder --> Business Unit folder --> Team folder
- **[exists]** Business Unit folder --> Environment folder --> Team folder
- Business Unit folder --> Team folder --> Environment folder
- Business Unit folder --> Team folder --> Flat - Projects with -dev, -prod suffix

We could also have the Netflix's tribe model which adds a notion of floating / non-permanent groups of people than can be moved easily, but not sure if it really would be in scope here.

I've seen all of those, but there might be others to consider.

**Comment by ludoo** [comment]
Hey Olivier, the examples are there to illustrate how to wire the individual components together, and how to use foundational setups for bootstrapping automation. So your third example above for example is just a matter of changing names, and does not warrant a new set of files.

Same for the fourth one, where projects are actual resources and need to "live" into separate, per environment setups which are applied (and segregated) using different service accounts created in the foundational phase.

Bottom line: we're not trying to offer ready-made setups for countless variations, but a selected few samples that match what we've seen in the field, allow for easy adaptation, and explain how to kickstart automation.
